pt_os_web_portal/backend/helpers/wifi_manager.py

```diff
@@ -106,21 +106,30 @@ def scan_and_get_results(self) -> List:
             WifiManager.wait_for(
                 self.is_scanning, "scan completion", condition_true=False, silent=True
             )
-
         logger.info("Scan completed")
-        results = self.wifi_interface.scan_results()
-        logger.info("ssids found: {}".format([r.ssid for r in results]))
-        return results
 
-    def connect(self, ssid: str, password: str) -> None:
+        networks: Dict[str, Any] = {}
+        for network in self.wifi_interface.scan_results():
+            ssid_to_display = (
+                network.ssid if network.freq < 5000 else f"{network.ssid} [5G]"
+            )
+            if (
+                ssid_to_display in networks
+                and getattr(networks[ssid_to_display], "signal", -100) > network.signal
+            ):
+                continue
+            networks[ssid_to_display] = network
+        logger.info(f"Found SSIDs: {tuple(networks.keys())}")
+        return [networks[ssid_to_display] for ssid_to_display in networks]
+
+    def connect(self, bssid: str, password: str) -> None:
         network_profile = None
         for r in self.scan_and_get_results():
-            if r.ssid == ssid:
+            if r.bssid == bssid:
                 network_profile = r
                 break
-
         if network_profile is None:
-            logger.info("Unable to find network matching SSID '%s'" % ssid)
+            logger.info(f"Unable to find network matching BSSID '{bssid}'")
             return
```

Comment:
returning in the connect method causes a positive response to POST /wifi-credentials, so in this error case or if we reach the end of the method without a connection being made, the UI incorrectly shows that the connection was successful.

obviously this issue is not introduced by this PR but is relevant so could be fixed here perhaps